misc/toml-parser/src/main/java/io/ballerina/toml/validator/SchemaValidator.java

```diff
@@ -88,26 +89,43 @@ public void visit(TomlTableNode tomlTableNode) {
                 DiagnosticInfo diagnosticInfo = new DiagnosticInfo("TVE0001", "error.unexpected.property",
                         DiagnosticSeverity.ERROR);
                 TomlDiagnostic diagnostic = new TomlDiagnostic(value.location(), diagnosticInfo,
-                        "Unexpected Property \"" + key + "\"");
+                        getUnexpectedPropertyErrorMessage(key));
                 tomlTableNode.addDiagnostic(diagnostic);
             }
         }
         for (String field : requiredFields) {
             DiagnosticInfo diagnosticInfo = new DiagnosticInfo("TVE0006", "error.required.field.missing",
                     DiagnosticSeverity.ERROR);
             TomlDiagnostic diagnostic = new TomlDiagnostic(tomlTableNode.location(), diagnosticInfo,
-                    "Missing required field \"" + field + "\"");
+                    getRequiredErrorMessage(field));
             tomlTableNode.addDiagnostic(diagnostic);
         }
     }
 
+    private String getRequiredErrorMessage(String field) {
+        Map<String, String> message = this.schema.message();
+        String typeCustomMessage = message.get(SchemaDeserializer.REQUIRED);
+        if (typeCustomMessage == null) {
+            return "missing required field '" + field + "'";
+        }
+        return typeCustomMessage.replace(PROPERTY_HOLDER, field);
+    }
+
+    private String getUnexpectedPropertyErrorMessage(String key) {
+        Map<String, String> message = this.schema.message();
+        String typeCustomMessage = message.get(SchemaDeserializer.ADDITIONAL_PROPERTIES);
+        if (typeCustomMessage == null) {
+            return "unexpected property '" + key + "'";
```

Comment:
Shall we rephrase this error message so that from the error message itself it's evident that the key/val pair is invalid because it's not in the schema? Maybe something like,
```
key 'foo' not supported in schema 'MySchema'
```